stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/messages/adapter/MessageListItemViewHolderFactory.kt

```diff
@@ -22,19 +24,45 @@ public open class MessageListItemViewHolderFactory {
-    public open fun createPlainTextViewHolder(parentView: ViewGroup): BaseMessageItemViewHolder<*> = MessagePlainTextViewHolder(parentView)
-    public open fun createReplyMessageViewHolder(parentView: ViewGroup): BaseMessageItemViewHolder<*> = createBaseMessageItemViewHolder(parentView)
-    public open fun createPlainTextWithAttachmentsViewHolder(parentView: ViewGroup): BaseMessageItemViewHolder<*> = createBaseMessageItemViewHolder(parentView)
-    public open fun createMediaAttachmentsViewHolder(parentView: ViewGroup): BaseMessageItemViewHolder<*> = OnlyMediaAttachmentsViewHolder(parentView)
-    public open fun createAttachmentsViewHolder(parentView: ViewGroup): BaseMessageItemViewHolder<*> = createBaseMessageItemViewHolder(parentView)
-    public open fun createLoadingIndicatorViewHolder(parentView: ViewGroup): BaseMessageItemViewHolder<*> = createBaseMessageItemViewHolder(parentView)
-    public open fun createThreadSeparatorViewHolder(parentView: ViewGroup): BaseMessageItemViewHolder<*> = createBaseMessageItemViewHolder(parentView)
+    public open fun createDateDividerViewHolder(parentView: ViewGroup): BaseMessageItemViewHolder<*> {
+        return DateDividerViewHolder(parentView)
+    }
+
+    public open fun createMessageDeletedViewHolder(parentView: ViewGroup): BaseMessageItemViewHolder<*> {
+        return MessageDeletedViewHolder(parentView)
+    }
+
+    public open fun createPlainTextViewHolder(parentView: ViewGroup): BaseMessageItemViewHolder<*> {
+        return MessagePlainTextViewHolder(parentView)
+    }
+
+    public open fun createReplyMessageViewHolder(parentView: ViewGroup): BaseMessageItemViewHolder<*> {
+        return createBaseMessageItemViewHolder(parentView)
+    }
+
+    public open fun createPlainTextWithAttachmentsViewHolder(parentView: ViewGroup): BaseMessageItemViewHolder<*> {
+        return createBaseMessageItemViewHolder(parentView)
+    }
+
+    public open fun createMediaAttachmentsViewHolder(parentView: ViewGroup): BaseMessageItemViewHolder<*> {
+        return OnlyMediaAttachmentsViewHolder(parentView)
+    }
+
+    public open fun createAttachmentsViewHolder(parentView: ViewGroup): BaseMessageItemViewHolder<*> {
+        return createBaseMessageItemViewHolder(parentView)
+    }
+
+    public open fun createLoadingIndicatorViewHolder(parentView: ViewGroup): BaseMessageItemViewHolder<*> {
+        return createBaseMessageItemViewHolder(parentView)
+    }
+
+    public open fun createThreadSeparatorViewHolder(parentView: ViewGroup): BaseMessageItemViewHolder<*> {
+        return createBaseMessageItemViewHolder(parentView)
+    }
 
     private fun createBaseMessageItemViewHolder(parentView: ViewGroup): BaseMessageItemViewHolder<MessageListItem> {
-        return object : BaseMessageItemViewHolder<MessageListItem>(parentView) {
-            override fun bindData(data: MessageListItem) = Unit
+        return object : BaseMessageItemViewHolder<MessageListItem>(View(parentView.context)) {
```

Comment:
Why do we need to create a new view instead of passing `parentView`?

Reply:
What's the point of attaching parentView to parentView? I suppose it will crash.

Reply:
I mean view holder view is attached to parentView by the adapter and they can't be the same

Reply:
We can create a meaningful view with "not implemented" text

Reply:
ah okay, it's `BaseMessageItemViewHolder`. Everything clear now